wp-content/themes/cf-voyager/lessons-cpt.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+/**
+ * Template Name: Lessons CPT
+ *
+ */
+
+get_header();
+
+$custom_terms = get_terms('lesson_types');
+
+foreach ($custom_terms as $custom_term) {
```

Comment:
What this `foreach` does is overwriting the $args at each iteration, so if you have 3 terms, the query at line 28 will get lessons having associated the third term.